cli/azd/extensions/microsoft.azd.extensions/internal/helpers.go

```diff
@@ -211,3 +264,29 @@ func AzdConfigDir() (string, error) {
 
 	return azdConfigDir, nil
 }
+
+// GlobArtifacts finds artifacts matching the given patterns
+func GlobArtifacts(patterns []string) ([]string, error) {
+	var allFiles []string
+
+	for _, pattern := range patterns {
+		// Check if the pattern is a concrete file path (no wildcards)
+		if !strings.Contains(pattern, "*") && !strings.Contains(pattern, "?") && !strings.Contains(pattern, "[") {
+			// It's a concrete file path, check if it exists
+			if _, err := os.Stat(pattern); err == nil {
+				allFiles = append(allFiles, pattern)
+			}
+			// If the file doesn't exist, skip it (no error to match glob behavior)
+			continue
+		}
+
+		// Use filepath.Glob for pattern matching
+		files, err := filepath.Glob(pattern)
+		if err != nil {
+			return nil, err
+		}
+		allFiles = append(allFiles, files...)
+	}
+
+	return allFiles, nil
+}
```

Comment:
Could these 2 cases be combined together to just use `filepath.Glob`?

Reply:
Thanks, I pushed some new changes - could you take a look at all the changes made in this PR and rewrite an appropriate PR description?